src/main/java/org/openstreetmap/atlas/checks/validation/areas/WaterbodyAndIslandSizeCheck.java

```diff
@@ -0,0 +1,336 @@
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(1,
+                            object.getOsmIdentifier(), surfaceAreaKilometers)));
+                }
+            }
+            // place=island or place=archipelago tags
+            if (IS_ISLAND.test(object))
+            {
+                if (surfaceAreaMeters < this.islandMinimumArea)
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(2,
+                            object.getOsmIdentifier(), surfaceAreaMeters)));
+                }
+                else if (surfaceAreaKilometers > this.islandMaximumArea)
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(THREE,
+                            object.getOsmIdentifier(), surfaceAreaKilometers)));
+                }
+            }
+            // place=islet tag
+            if (IS_ISLET.test(object))
+            {
+                if (surfaceAreaMeters < this.isletMinimumArea)
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(FOUR,
+                            object.getOsmIdentifier(), surfaceAreaMeters)));
+                }
+                else if (surfaceAreaKilometers > this.isletMaximumArea)
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(FIVE, object.getOsmIdentifier())));
+                }
+            }
+        }
+        // There are two Relations to examine:
+        // 1. Islands included as members of a MultiPolygon
+        // 2. Water bodies included as members of a Multipolygon
+        // Each Relation will have n numbers of instructions, dependent on the amount of members
+        // that fail the size check
+        else if (object instanceof Relation)
+        {
+            final Relation relation = (Relation) object;
+            // Remove non-Area members && entities that have already been examined
+            final Set<RelationMember> relationMembers = relation.members().stream()
+                    .filter(member -> member.getEntity() instanceof Area
+                            && !this.isFlagged(member.getEntity().getIdentifier()))
+                    .collect(Collectors.toSet());
+
+            if (!relationMembers.isEmpty())
+            {
+                // Get valid members that fail the surface area size check
+                final Set<RelationMember> invalidPolygonMembers = this
+                        .invalidPolygonMembers(relationMembers);
+
+                if (!invalidPolygonMembers.isEmpty())
+                {
+                    // Get instructions for each member
+                    final List<String> instructions = this
+                            .getMemberInstructions(invalidPolygonMembers);
+
+                    if (!invalidPolygonMembers.isEmpty())
```

Comment:
You can remove this conditional because you are already within a conditional block that checked that the invalidPolygonMembers wasn't empty (ref. line 207).

Reply:
Yeah, that should be `instructions.isEmpty`